torch/_higher_order_ops/cond.py

```diff
@@ -157,11 +157,18 @@ def trace_cond(proxy_mode, func_overload, pred, true_fn, false_fn, operands):
     pre_dispatch = getattr(proxy_mode, "pre_dispatch", False)
 
     with disable_proxy_modes_tracing():
+        # We'll use the current decomposition table to make sure operatos in subgraphs are
+        # decomposed properly.
+        decomp_table = torch.fx.experimental.proxy_tensor.CURRENT_DECOMPOSITION_TABLE
```

Comment:
do we need this for map/while_loop? If so we should probably have a reusable `reenter_make_fx` that does this

Reply:
Yeah! I'll do it as a follow up PR. Thanks for pointing it out!